frontend/src/components/LogViewer.tsx

```
@@ -85,17 +85,47 @@ const LogViewer: React.FC<LogViewerProps> = ({
+        return 'text-gray-400'
       default:
-        return 'white'
+        return 'text-gray-200'
+    }
+  }
+
+  // Clean ANSI color codes and extract code path to show at the end
+  const cleanLogMessage = (message: string, level: string, timestamp: string) => {
+    let cleaned = message
+      // Remove ANSI color codes like [0m, [32m, [38;5;8m, etc.
+      .replace(/\x1b\[[0-9;]*m/g, '')
```

Biome (1.9.4)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)